### PR TITLE
CRAB-28961: Csharp connector SDK using nupkg 

### PR DESCRIPTION
**Pull request by mar1u50**
This is complementary to https://bitbucket.org/seeq12/crab/pull-requests/16541 and it changes Connector SDK for csharp to use nupkg libraries from nuget.

We have no nupkg packages on nuget yet but for testing this PR we can download them from https://teamcity.seeq-labs.com/buildConfiguration/CRAB_WindowsConnectorSdk/204759?buildTab=artifacts#%2Fconnector-sdk%2Fnupkg
and then we can add them in a local nupkg repo using 
```
nuget add Seeq.Link.SDK.57.0.0.nupkg -source /c/nuget/
nuget add Seeq.SDK.57.0.0.nupkg -source /c/nuget/
nuget add Seeq.Utilities.57.0.0.nupkg -source /c/nuget/
nuget add Seeq.Link.Agent.57.0.0.nupkg -source /c/nuget/
```

Then you need to add /c/nuget as a nuget source (easiest way is to do it in Visual Studio Tools >Nuget Package Manager > Settings):
image

### CODE REVIEW

File: clean.bat [hunk 1]
```diff
@@ -0,0 +1,4 @@
+echo y|rmdir /s MyCompany.Seeq.Link.Connector.MyConnector\bin 
+echo y|rmdir /s MyCompany.Seeq.Link.Connector.MyConnector\obj
+echo y|rmdir /s c:\seeq-connector-sdk\csharp-new\Seeq.Link.SDK.Debugging.Agent\bin 
+echo y|rmdir /s c:\seeq-connector-sdk\csharp-new\Seeq.Link.SDK.Debugging.Agent\obj
```
Comment (AlexDorobantiu) [inline clean.bat:4]: We may want to consider whether or not the clean should delete also nuget and/or the packages folder.

**Review by AlexDorobantiu — APPROVED** [review]

File: clean.bat [hunk 2]
```diff
@@ -0,0 +1,4 @@
+echo y|rmdir /s MyCompany.Seeq.Link.Connector.MyConnector\bin 
+echo y|rmdir /s MyCompany.Seeq.Link.Connector.MyConnector\obj
+echo y|rmdir /s c:\seeq-connector-sdk\csharp-new\Seeq.Link.SDK.Debugging.Agent\bin
```
[the remark this answers is not in the document]
Comment (mar1u50) [reply clean.bat:3]: good point. Thx for spotting this

File: clean.bat (hunk 1 above)
Comment (mar1u50) [reply clean.bat:4]: I see no reason to remove the packages folder. This usually comes from nuget and cannot change.

**Comment by mar1u50** [comment]
I tried to switch to PackageReference because is is newer, better, etc. All works well from VS but from command line the project wont build with msbuild v4.0.30319. dotnet, nuget, msbuild restore all will fail when using 30319. I reverted to use package.config for now. I do not think we want to switch to vs compiler right now

**Review by DJamesP — APPROVED** [review]

**Review by DJamesP — COMMENTED** [review]
The PackageReference didn't work?

**Comment by mar1u50** [comment]
> The PackageReference didn't work?

It works from Visual Studio, but form command line: 1) it works if we use Microsoft Build Tools 2015, 2) It works if we use the compiler from Visual Studio 3) It doesn't work with the MSBuild we currently require (v4.0.30319).

If we can require Microsoft Build Tools 2015 to be installed, all will be fine.

What is failing: "nuget restore" from command line.

**Pull request merged**